client/ayon_deadline/addon.py

```diff
@@ -38,6 +38,11 @@ def initialize(self, studio_settings):
 
         self.deadline_servers_info = deadline_servers_info
 
+        self._pools_per_server = {}
+        self._limit_groups_per_server = {}
+        self._groups_per_server = {}
+        self._machines_per_server = {}
```

Comment:
I wonder if we should make these class variables instead. I suppose having them here on `initialize` would mean that on each reset of the publisher UI it would still not hit the cache because a new `AddonsManager` instance gets created and hence, the addon is initialized again? Or is the addon itself cached in the session too? (I'm not sure?)